workers_node_groups.tf

```diff
@@ -0,0 +1,176 @@
+resource "aws_iam_role" "managed_node_groups" {
+  count                 = local.worker_group_managed_node_group_count > 0 ? 1 : 0
+  name                  = "${var.workers_role_name != "" ? var.workers_role_name : aws_eks_cluster.this.name}-managed-node-groups"
+  assume_role_policy    = data.aws_iam_policy_document.workers_assume_role_policy.json
+  permissions_boundary  = var.permissions_boundary
+  path                  = var.iam_path
+  force_detach_policies = true
+  tags                  = var.tags
+}
+
+resource "aws_iam_role_policy_attachment" "managed_node_groups_AmazonEKSWorkerNodePolicy" {
+  count      = local.worker_group_managed_node_group_count > 0 ? 1 : 0
+  policy_arn = "arn:aws:iam::aws:policy/AmazonEKSWorkerNodePolicy"
+  role       = aws_iam_role.managed_node_groups[0].name
+}
+
+resource "aws_iam_role_policy_attachment" "managed_node_groups_AmazonEKS_CNI_Policy" {
+  count      = local.worker_group_managed_node_group_count > 0 ? 1 : 0
+  policy_arn = "arn:aws:iam::aws:policy/AmazonEKS_CNI_Policy"
+  role       = aws_iam_role.managed_node_groups[0].name
+}
+
+resource "aws_iam_role_policy_attachment" "managed_node_groups_AmazonEC2ContainerRegistryReadOnly" {
+  count      = local.worker_group_managed_node_group_count > 0 ? 1 : 0
+  policy_arn = "arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryReadOnly"
+  role       = aws_iam_role.managed_node_groups[0].name
+}
+
+resource "aws_iam_role_policy_attachment" "managed_node_groups_additional_policies" {
+  count      = local.worker_group_managed_node_group_count > 0 ? length(var.workers_additional_policies) : 0
```

Comment:
This would be a really good place to use `for_each` instead of `count` so that reordering or removing a policy doesn't cause it to re-attach every single one. You can [find more info here](https://www.terraform.io/docs/configuration/resources.html#using-sets).

It would probably end up looking something like this:
```hcl
resource "aws_iam_role_policy_attachment" "managed_node_groups_additional_policies" {
  for_each = toset(var.workers_additional_policies)
  
  role       = aws_iam_role.managed_node_groups[0].name
  policy_arn = each.key
}
```

I suspect you can replace almost every instance of `count` with something like this, however it's less important for cases when at most there where be one entity.

Reply:
This is much simpler. I will make this change though it will differentiate the code of the managed node groups from the rest of the worker groups in terms of code